src/faultlog/faultlog_main.cpp

```diff
@@ -25,6 +26,111 @@ using Severity = sdbusplus::xyz::openbmc_project::Logging::server::Entry::Level;
 
 using Binary = std::vector<uint8_t>;
 using PropVariant = sdbusplus::utility::dedup_variant_t<Binary>;
+
+/** @brief Helper method to create faultlog pel
+ *
+ *  @param[in] bus - D-Bus to attach to
+ *  @param[in] guardRecords - hardware isolated records to parse
+ *  @param[in] hostPowerOn - flag to check if called during host IPL
+ */
+void createNagPel(sdbusplus::bus::bus& bus,
+                  const GuardRecords& unresolvedRecords, bool hostPowerOn)
```

Comment:
`hostPowerOn` can be made const to be on more safer side. Just a suggestion

Reply:
it is by value